### ISSUE TITLE
Initial displayed message disappeared

### ISSUE THREAD

**Issue by MfMartel** [alan-s-lee/OnPoint#5]
Hi,

I have some questions about the current version of the index.js. I am struggling with the `moveCursor()` function: the Initial displayed message does not appear anymore once we launch the game, instead, there is just the center circle. 

I noticed that if I comment out l. 612-614:  ` if(trial == 0){moveCursor();}`, then the initial text with the instructions does appear, but pressing space does not do anything. I need to first click with the mouse and then press any key (and not just the space bar) and it works and the experiment can start.

I have never used js before so I have some trouble localizing and fixing this particular issue. Any chance you know what this is about?

Many thanks,
Marie

**Comment by alan-s-lee** [comment]
Hey Marie, 

It seems like we had a bug in our code previously that skipped out on the initial message, sorry about that! It should be fixed now with the most recent push. 

To quickly answer your question about what went wrong, in this previous version where you ran into the issue the game skipped it's phase from displaying the initial message to the "search phase" directly (which is the center circle), so we just had to restructure the game flow a bit to make sure we are not skipping the "between blocks" game phase were we display messages in the very beginning. Hope this provides some context as to what went wrong! 

I will be closing the issue as it should be fixed with this most recent code change, but please feel free to reopen it if it does not resolve your issue!

Best,
Alan

**Comment by MfMartel** [comment]
Hi Alan,

Thank you for your help and the extra info.
The initial message is now there but there is still the issue that if I press space as indicated it does not do anything. And I have to click on the mouse and then space works. 
I assume that there is something wrong with the way the program recognises keydown? maybe in the `lockChangeAlert()`? but I tried commenting/uncommenting lines and it messes up things more without fixing the space issue. Or maybe in the sequence of events? But I couldn't find in the script where it is specified that it waits for a click...

Best,
Marie

**Comment by alan-s-lee** [comment]
Hey Marie,

I am personally not running into the same issue when testing it on my local setup (space works as intended without a need to click on the mouse). Though there are a few things we can try to make sure everything is working as intended.

1. Make sure you do not have any additional changes beyond the github code. If you wish to retain some modifications you have, you can also try creating a new directory in your computer to pull the github code to test it out.

2. When you test things out in localhost, try pulling out "inspect elements" on the side (for on chrome try "cmd" + "options" + "i" or refer to https://developer.chrome.com/docs/devtools/shortcuts/). Then on the panel that pops up on your webpage select the "console" tab. Proceed to the experiment with this open, when the experiment starts check for the following entries in the "console".
 - "enter1"
 - "Initial X: 0 Initial Y:0"
 - "Attempted to lock pointer"
 - "The pointer lock status is now locked"
The important thing is to make sure that last entry is there, to determine whether the pointer lock is working as intended. Please let me know what the results are when you try this both with your current setup and when you try it purely on the github code! 

To provide more context if you would like to also try debugging on your own, the pointer lock is the feature we added so that participants can complete the reaching experiment without needing to return to the center every reach. This is done by locking the mouse position and creating a "fake cursor" that can be teleported near the center after each reach. In the beginning of the experiment, `line 533` should trigger the `setPointerLock()` function, which attempts to lock the pointer. This causes the listener that we created in `line 510` to run `lockChangeAlert()`, since every time there is a change in the pointer lock status will call `lockChangeAlert()`. Inside this function, if the pointer lock is turned on, we then create listeners that run specific functions when there is a `keydown` or `mousemove`. `keydown` would be the action that corresponds to our initial space press, which will run the `advance_block()` function. In that, since the `bb_mess` variable would be `0`, the first case will be triggered to begin the `search_phase()` function, which shows the start circle.

I have reopened the issue so that we can figure this out, but please let us know what the "console" shows you when you start the experiment!

Best,
Alan

**Comment by MfMartel** [comment]
Hi Alan,

Thanks! Sorry for not specifying it, but yes I am running your code as it is (I assumed that I needed first to have it working and then I could put back the changes I made).

I tried what you asked and on the console I get:
`Request for pointer lock was denied because Element.requestPointerLock() was not called from inside a short running user-generated event handler, and the document is not in full screen.`

One thing though, that seems to be relevant: I just tried it out on a different browser since I'm not using Chrome and space works well on Edge but not on Firefox (which is what I'm using). In the original version of the code though, it was working well on Firefox, so I assume something still changed compared to before?
(And the console on Edge gets "The pointer lock status is now locked" as it should be)

Best,
Marie

**Comment by alan-s-lee** [comment]
Hey Marie,

Thank you for the clarification! 

The only change we've made since the previous version that worked on Firefox is we now use the [Pointer Lock API](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_Lock_API). From the console message you've provided, it seems like Firefox is not liking the new API we included. If you do not need the new feature that no longer requires participants to "return to the center" you can still access the old versions on github by looking at previous commits (before 5/20/2020). Otherwise, we do recommend using Chrome or Edge since those are the platforms we used during development! 

Hope that answers you're question, and we apologize for the inconvenience as browsers all have different levels of compatibility with certain API's. Please close this issue if you're satisfied with the answer, but let us know if there's anything else we can help with! 

Best,
Alan

**Comment by MfMartel** [comment]
Hi,

No worries, thank you for letting me know, I had missed the recommendation about browsers, my bad.

Apologies for the late reply, I tried it on Chrome and I had some weird behaviours (e.g. message and experiment not in the middle of the screen, no message appearing when I pressed Escape to quit fullscreen in the middle of the experiment) so I was trying to figure things out. But these behaviors are not stable, sometimes it is happening and sometimes it is not so I assume it is more an issue of me modifying index files and then the Cache being not empty.

My last try this morning after emptying and cleaning all the cache/history showed no issue at all. So I think I can now close the issue. Thanks again for your help.

Best,
Marie
